api/environments/models.py

```diff
@@ -358,6 +375,187 @@ def get_feature_state(
             )
         )
 
+    def get_metrics_payload(self, with_workflows: bool = False) -> EnvMetricsPayload:
+        """
+        Returns total feature count and enabled-by-default feature count
+        scoped to this environment's project.
+        """
+        from edge_api.identities.models import EdgeIdentity
+
+        features_qs = self._get_features_metrics_queryset(with_workflows=with_workflows)
+
+        _features_aggregation_result: dict[str, int] = {}
+
+        # Closure to avoid recalculating twice the features metrics
+        def get_feature_agg(key: str) -> int:
+            nonlocal _features_aggregation_result
+            if not _features_aggregation_result:
+                _features_aggregation_result = features_qs.aggregate(
+                    total=models.Count("id"),
+                    enabled=models.Count("id", filter=Q(enabled=True)),
+                )
+            return _features_aggregation_result[key]
```

Comment:
Is this a caching strategy? Can you expand on why we need it? I'd love to help flattening this into an easier to read version, if it's at all possible.